app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java

```diff
@@ -323,6 +326,70 @@ public static BgReading create(double raw_data, double filtered_data, Context co
                 }
 
                 bgReading.calculated_value = Math.min(400, Math.max(40, bgReading.calculated_value));
+                Log.w(TAG, "create: NEW VALUE CALCULATED AT: " + bgReading.calculated_value);
+
+                bgReading.save();
+                bgReading.perform_calculations();
+                Notifications.getInstance(context).notificationSetter(context);
+                BgSendQueue.addToQueue(bgReading, "create", context);
+            }
+        }
+        Log.w("BG GSON: ",bgReading.toS());
+
+        return bgReading;
+    }
+    public static BgReading create(double raw_data, Context context, Long timestamp) {
```

Comment:
If we have a function that is never called we should remove it...

On Tue, May 19, 2015 at 11:58 PM, Stephen Black <notifications@github.com>
wrote:

> In app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java
> <https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/23#discussion_r30646134>
> :
>
> > @@ -323,6 +326,70 @@ public static BgReading create(double raw_data, double filtered_data, Context co
> >                  }
> >
> >                  bgReading.calculated_value = Math.min(400, Math.max(40, bgReading.calculated_value));
> > +                Log.w(TAG, "create: NEW VALUE CALCULATED AT: " + bgReading.calculated_value);
> > +
> > +                bgReading.save();
> > +                bgReading.perform_calculations();
> > +                Notifications.getInstance(context).notificationSetter(context);
> > +                BgSendQueue.addToQueue(bgReading, "create", context);
> > +            }
> > +        }
> > +        Log.w("BG GSON: ",bgReading.toS());
> > +
> > +        return bgReading;
> > +    }
> > +    public static BgReading create(double raw_data, Context context, Long timestamp) {
>
> the existing one is the one that takes filtered data as a param, this
> function we are commenting on now is one that does not take filtered data
> and nothing actually ever calls this
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/23/files#r30646134>
> .
>